CMakeLists.txt

```diff
@@ -1,61 +1,44 @@
 # ensmallen CMake configuration.  This project has no configurable options---it
 # just installs the headers to the install location, and optionally builds the
 # test program.
-cmake_minimum_required(VERSION 2.8.10)
-project(ensmallen C CXX)
+cmake_minimum_required(VERSION 3.3.2)
+project(ensmallen 
+        VERSION 2.10.5
+        LANGUAGES C CXX)
 
+# Configurable options for CMake
 option(USE_OPENMP "If available, use OpenMP for parallelization." ON)
 option(BUILD_TESTS "Build tests." ON)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMake")
 
-# Ensure we have C++11 features.  Since we support CMake < 3.1, this needs a
-# little bit of special handling.
-if ((${CMAKE_MAJOR_VERSION} LESS 3 OR
-    (${CMAKE_MAJOR_VERSION} EQUAL 3 AND ${CMAKE_MINOR_VERSION} LESS 1))
-    AND NOT FORCE_CXX11)
-  # Older versions of CMake do not support target_compile_features(), so we have
-  # to use something kind of hacky.
-  include(CMake/CXX11.cmake)
-  check_for_cxx11_compiler(HAS_CXX11)
-  if(NOT HAS_CXX11)
-    message(FATAL_ERROR "No C++11 compiler available!")
+# Set required C++ standard to C++11 
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Create library target
+add_library(ensmallen INTERFACE)
+target_include_directories(ensmallen INTERFACE
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
```

Comment:
Are you sure that we need the `ensmallen` library target?  There's actually no library that needs to be built, only the header files.  Maybe there is something that I am missing?

Reply:
In modern CMake, we make targets out of pretty much every concrete output, be it libraries or executables. This applies to even header only libraries, since we are still providing some output - our library, in the form of it's header files. As you can see, I used the keyword `INTERFACE` for the library. An `INTERFACE` library target does not directly create build output, but it may have properties set on it and it may be installed, exported and imported, which is what we need for our header only library.

Reply:
I see!  Thanks for the explanation.  I went and did a little reading about `INTERFACE` libraries too.  I guess I will see how this works if we go through and import ensmallen as a target for mlpack. :)